src/wp-includes/class-wp-scripts.php

```diff
@@ -886,10 +939,11 @@ public function add_data( $handle, $key, $value ) {
 	 * @since 6.3.0
 	 * @see WP_Scripts::should_delay_inline_script()
 	 *
+	 * @param string[] $handles Handles to check.
 	 * @return bool True if the inline script present, otherwise false.
 	 */
-	public function has_delayed_inline_script() {
-		foreach ( $this->registered as $handle => $script ) {
```

Comment:
Note: I noticed this was looping over _all_ registered scripts, when in reality it should only be looping over the scripts that are enqueued (i.e. in `WP_Scripts::$to_do`). This is now done here.

Reply:
Test case b9543ac1855932060094690a2878adc57f316b64 captures this.